### PR TITLE
Adds basic features.

### PR DESCRIPTION
**Pull request by Jelle-S**
- [x] taskSymlinkFolderFileContents
- [x] taskScp
- [x] taskPushPackage
- [ ] taskDatabaseBackup (see https://github.com/hechoendrupal/drupal-console/issues/3141)
- [ ] taskSwitchCode (should we create a dedicated task for this when we have http://robo.li/tasks/Filesystem/#filesystemstack?)
- [ ] taskClearOpcache (how should we do this? see https://ma.ttias.be/how-to-clear-php-opcache/)
- [ ] taskCleanServer (should we create a dedicated task for this when we have http://robo.li/tasks/Filesystem/#filesystemstack?)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "prefer-stable": true,
     "require": {
         "consolidation/robo": "~1.0",
-        "php": ">=5.5.0"
+        "php": ">=5.5.0",
+        "phpseclib/phpseclib": "^2.0"
```
Comment (daften) [inline composer.json]: No PECL_SSH2
so_good

File: tests/PushPackageTest.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+        // Mock the SCP adapter.
+        $adapter = $this->mockScpAdapter($host, $auth, $port, $timeout);
+        $adapter
+            ->expects($this->once())
+            ->method('put')
+            ->with($destinationFolder, $localFile)
+            ->willReturn(true);
+
+
```
Comment (daften) [inline tests/PushPackageTest.php]: Unneeded white line

File: src/SymlinkFolderFileContents.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+namespace DigipolisGent\Robo\Task\Deploy;
+
+use Robo\Collection\CollectionBuilder;
+use Robo\Contract\BuilderAwareInterface;
+use Robo\Task\BaseTask;
+use Robo\Task\Filesystem\FilesystemStack;
+use Symfony\Component\Finder\Finder;
+
+class SymlinkFolderFileContents extends BaseTask implements BuilderAwareInterface
```
Comment (daften) [inline src/SymlinkFolderFileContents.php]: So the goal is to call a robo command via an ssh command to do the deploy. Can instructions for this be added in the README already? :)

File: src/Scp.php [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    {
+        $this->stopOnFail = $stopOnFail;
+
+        return $this;
+    }
+
+    /**
+     * Set the ScpFactory.
+     *
+     * @param sting|\DigipolisGent\Robo\Task\Deploy\Scp\Factory\ScpFactoryInterface $class
```
Comment (daften) [inline src/Scp.php]: sting -> string

File: src/Ssh.php [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    {
+        $this->stopOnFail = $stopOnFail;
+
+        return $this;
+    }
+
+    /**
+     * Set the SshFactory.
+     *
+     * @param sting|\DigipolisGent\Robo\Task\Deploy\Ssh\Factory\SshFactoryInterface $class
```
Comment (daften) [inline src/Ssh.php]: sting -> string

File: src/PushPackage.php [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+use Robo\Task\BaseTask;
+
+class PushPackage extends BaseTask implements BuilderAwareInterface
+{
+    use loadTasks;
+    use \Robo\Task\Remote\loadTasks;
+    use \Robo\TaskAccessor;
+
+    /**
+     * The server to scp to/from.
```
Comment (Jelle-S) [inline src/PushPackage.php]: Well, there's get and put, so get from the server, put (upload) to the server.

File: src/PushPackage.php [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+    public function scpFactory($class)
+    {
+        if (!is_subclass_of($class, ScpFactoryInterface::class)) {
+            throw new InvalidArgumentException(sprintf(
+                'SCP Factory %s does not implement %s.',
+                $class,
+                ScpFactoryInterface::class
+            ));
+        }
+        $this->scpFactory = $class;
```
Comment (Jelle-S) [inline src/PushPackage.php]: All code using the factory should work with both a string and object: is_subclass_of works and call_user_func_array works with either strings or objects.

File: src/PushPackage.php (hunk 6 above)
Comment (daften) [inline src/PushPackage.php]: Yes, in the separate files, but this class only _pushes_ the package to the server AFAICS

Comment (Jelle-S) [inline src/PushPackage.php]: Ah, right, that is correct.

**Pull request merged**